lib/src/main/java/space/taran/arklib/utils/ImageUtils.kt

```diff
@@ -77,42 +62,56 @@ object ImageUtils {
         placeHolder: Int,
         view: ImageView
     ) {
-        Log.v(LOG_PREFIX, "loading image $image")
+        val signature = "$id$THUMBNAIL_SIGNATURE"
 
-        Glide.with(view.context)
-            .load(image?.toFile())
-            .placeholder(placeHolder)
-            .signature(ObjectKey("$id$THUMBNAIL_SIGNATURE"))
-            .transition(DrawableTransitionOptions.withCrossFade())
-            .into(view)
+        view.load(image?.toFile(), arkImageLoader) {
+            scale(Scale.FILL)
+            placeholder(placeHolder)
+            diskCacheKey(signature)
+            memoryCacheKey(signature)
+            crossfade(true)
+            logListener("[Thumbnail]", id, image)
+        }
     }
 
-    fun <T> glideExceptionListener() = object : RequestListener<T> {
-        override fun onLoadFailed(
-            e: GlideException?,
-            model: Any?,
-            target: Target<T>?,
-            isFirstResource: Boolean
-        ): Boolean {
-            Log.w(
-                LOG_PREFIX,
-                "load failed with message: ${
-                e?.message
-                } for target of type: ${
-                target?.javaClass?.canonicalName
-                }"
-            )
-            return true
-        }
+    fun ImageRequest.Builder.logListener(
+        prefix: String,
+        id: ResourceId,
+        image: Path?
+    ) {
+        listener(
+            onStart = {
+                Log.d(LOG_PREFIX, "$prefix Start loading path[$image] id[$id]")
+            },
+            onCancel = {
+                Log.d(LOG_PREFIX, "$prefix Cancel loading path[$image] id[$id]")
+            },
+            onError = { _, result ->
+                Log.d(
+                    LOG_PREFIX,
+                    "$prefix Error[${result.throwable}] when loading path[$image] id[$id]"
+                )
+            },
+            onSuccess = { _, result ->
+                Log.d(LOG_PREFIX, "$prefix Loaded path[$image] id[$id]")
+            },
+        )
+    }
 
-        override fun onResourceReady(
-            resource: T,
-            model: Any?,
-            target: Target<T>?,
-            dataSource: DataSource?,
-            isFirstResource: Boolean
-        ) = false
+    val arkImageLoader by lazy {
+        ImageLoader.Builder(app)
+            .components {
+                if (SDK_INT >= 28) {
```

Comment:
Let's use constant API.

```suggestion
                if (SDK_INT >= Build.VERSION_CODES.P) {
```

Reply:
It’s more convenient for me to read numbers, but okay